src/main/java/at/ac/tuwien/damap/repo/GdprRepo.java

```diff
@@ -0,0 +1,40 @@
+package at.ac.tuwien.damap.repo;
+
+import at.ac.tuwien.damap.rest.gdpr.domain.GdprResult;
+import at.ac.tuwien.damap.rest.gdpr.domain.HqlQuery;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import javax.persistence.EntityManager;
+import java.util.*;
+
+@ApplicationScoped
+public class GdprRepo {
+
+    @Inject
+    EntityManager entityManager;
+
+    public List<GdprResult> getGdprDataByUniversityId(List<HqlQuery> queries, String universityId) {
+
+        List<GdprResult> results = new ArrayList<>();
+        for (HqlQuery query : queries) {
+            results.add(this.getGdprEntityDataByUniversityId(query, universityId));
+        }
+        return results;
+    }
+
+    private GdprResult getGdprEntityDataByUniversityId(HqlQuery query, String universityId) {
+
+        GdprResult gdprResult = new GdprResult();
+        gdprResult.setEntity(query.getEntityName());
+
+        @SuppressWarnings("unchecked")
+        List<Map<String, Object>> result = entityManager.createQuery(query.getHql(), (Class<Map<String, Object>>) (Class<?>) Map.class)
+                .setParameter("id", universityId).getResultList();
+        result.forEach(map -> map.values().removeIf(Objects::isNull));
```

Comment:
please add a comment detailing the behavior of this removal, as on first glance i was worried we might be removing key information here (like personID, which is relevant result itself)
```suggestion
        //check if any of the principal output/result entities (e.g. Access, Consent, Contributor) are empty, and remove them if so.
```